### PR TITLE
Version checker

### LINKED ISSUE #345: Version checker
Would be nice if ASB Explorer would automatically check for latest version and tell me if a new version is available. As a later update, it would be nice if it would also auto-update. I want to try and add the versioning support and so want a simple issue to track

### PR DESCRIPTION
**Pull request by rajrao**
Fixes #345 
When a newer version is detected, then a link is displayed on the Main Form
image

Clicking on the link shows the following dialog
image

### CODE REVIEW

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-PullRequest0347.113.build.567 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/24412935) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/7c64bc09b2 by @)

**Comment by ErikMogensen** [comment]
This will very useful. 

A couple of things:

- Service Bus Explorer is blueish so please change the New version available link to dark blue. 
- I think you should add the version number of the latest release to the above link
- The above link should be Anchored Top,Right. Otherwise it may disappear if you resize the window.
- Add the copyright text to GitHubReleaseProvider.cs just like you did on the other file you added.

File: src/ServiceBusExplorer/Forms/MainForm.cs [hunk 1]
```diff
@@ -279,6 +279,15 @@ public MainForm()
             GetServiceBusNamespaceSettingsFromConfiguration();
             ReadEventHubPartitionCheckpointFile();
             UpdateSavedConnectionsMenu();
+
```
Comment (ErikMogensen) [inline src/ServiceBusExplorer/Forms/MainForm.cs]: Please extract these lines to a method  to keep the MainForm constructor on as few lines as possible. That will make it easier to read.

File: src/ServiceBusExplorer/Forms/MainForm.cs [hunk 2]
```diff
@@ -279,6 +279,15 @@ public MainForm()
...
+            if (!VersionProvider.IsLatestVersion(out var releaseInfo))
+            {
+                linkLabelNewVersionAvailable.Visible = true;
+            }
+            else
+            {
+                linkLabelNewVersionAvailable.Visible = false;
+            }
```
Comment (ErikMogensen) [inline src/ServiceBusExplorer/Forms/MainForm.cs]: Please extract these lines to a method to keep the MainForm constructor on as few lines as possible. That will make it easier to read.

Comment (SeanFeldman) [reply src/ServiceBusExplorer/Forms/MainForm.cs]: Also, there's no need to do `if/else`. You get a boolean, which can be assigned to `linkLabelNewVersionAvailable.Visible` in one liner. Maybe then @ErikMogensen will be ok not to have a helper method 😀

Comment (rajrao) [reply src/ServiceBusExplorer/Forms/MainForm.cs]: image

image

Done!

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-PullRequest0347.114.build.568 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/24441401) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/255ce219a3 by @)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-PullRequest0347.115.build.569 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/24479688) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/99914fd2fc by @rajrao)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-PullRequest0347.116.build.570 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/24479800) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/f7b3b29a43 by @)

**Comment by paolosalvatori** [comment]
Thanks for the PR @rajrao, I merged your changes into the code :)

**Pull request merged**